pkg/apis/apps/types.go

```diff
@@ -187,6 +187,12 @@ type StatefulSetStatus struct {
 	// updateRevision, if not empty, indicates the version of the StatefulSet used to generate Pods in the sequence
 	// [replicas-updatedReplicas,replicas)
 	UpdateRevision string
+
+	// collisionCount is the count of hash collisions for the StatefulSet. The StatefulSet controller
+	// uses this field as a collision avoidance mechanism when it needs to create the name for the
+	// newest ControllerRevision.
+	// +optional
+	CollisionCount *int64
```

Comment:
@thockin 
> JSON is known to have trouble carrying int64 values, which can make automated API fuzzing difficult.

This is not an issue that users have actually reported having, and fuzzing currently works for these API objects. If you configure your decoder appropriately or use structs, or classes depending on the language, to decode, disambiguating int64 values is definitely achievable without much work.

Given that the API for [Deployment](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/deployment.md) and [DeamonSet](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/daemonset-update.md) both went through community design review prior to implementation and merge, that there is no measurable performance benefit to using an int32 in this case, that we use int64 in many other places in the API, and that we don't have an actual issue to motivate changing all the workloads API objects, I don't see why we should switch this field to int32.